### PR TITLE
rose task-env: obey calendar mode

### LINKED ISSUE #1842: rose task-env: obey cylc calendar mode
`rose task-env` does not obey the cylc cycling mode (calendar mode) when calculating cycle offsets.

For example, the following suite:
```ini
[scheduling]
    cycling mode = 360day
    initial cycle point = 1997-02-27
    final cycle point = 1997-03-01
    [[dependencies]]
        [[[P1D]]]
            graph = print_rose_date
[runtime]
    [[print_rose_date]]
        env-script = eval $(rose task-env --cycle-offset=P1D)
        script = diff <(rose date -c -s -P1D) <(sed "s@.*/@@" <<<$ROSE_DATACP1D)
```
fails on 1997-02-30 and 1997-03-01.

### PR DESCRIPTION
**Pull request by matthewrmshin**
rose task-env was not working correctly with alternate calendars like
360day, etc. This change fixes #1842.

@benfitzpatrick @arjclark please review.

### CODE REVIEW

File: lib/python/rose/suite_engine_proc.py [hunk 1]
```diff
@@ -511,49 +525,58 @@ def get_tasks_auths(self, suite_name):
...
-        """
-
-        t = TaskProps()
+        """Return a TaskProps object containing suite task's attributes."""
+        calendar_mode = self.date_time_oper.get_calendar_mode()
+        try:
+            return self._get_task_props(*args, **kwargs)
+        finally:
+            # Restore calendar mode if changed
+            self.date_time_oper.set_calendar_mode(calendar_mode)
```
Comment (matthewrmshin) [inline lib/python/rose/suite_engine_proc.py]: Wrap to allow calendar mode to swap temporarily.

File: lib/python/rose/suite_engine_proc.py [hunk 2]
```diff
@@ -511,49 +525,58 @@ def get_tasks_auths(self, suite_name):
...
         # are defined as well.
-        if t.suite_name is not None and t.task_id is not None:
-            return t
+        if tprops.suite_name is not None and tprops.task_id is not None:
+            return tprops
 
-        t = self.get_task_props_from_env()
+        tprops = self.get_task_props_from_env()
+        # Modify calendar mode, if possible
+        self.date_time_oper.set_calendar_mode(tprops.cycling_mode)
```
Comment (matthewrmshin) [inline lib/python/rose/suite_engine_proc.py]: Modify calendar mode, if necessary.

File: lib/python/rose/date.py [hunk 3]
```diff
@@ -379,7 +393,7 @@ def _convert_duration(date_time_oper, opts, args):
     """Implement usage 3 of "rose date", convert ISO8601 duration."""
     time_in_8601 = date_time_oper.duration_parser.parse(args[0])
     time = time_in_8601.get_seconds()
-    options = {'S': time, 'M': time/60, 'H': time/3600}
+    options = {'S': time, 'M': time / 60, 'H': time / 3600}
```
Comment (matthewrmshin) [inline lib/python/rose/date.py]: Syntax test battery fails here (on master). I also raised #1838 for this (and other stuffs).

**Comment by matthewrmshin** [comment]
Sorry I have mixed pylint improvement (for `rose.suite_engine_proc`) with this change, but I have now highlighted the essential changes in my comments.

**Comment by matthewrmshin** [comment]
Branch rebased.

**Comment by arjclark** [comment]
Review 2 looks good to me (once I knew where I was looking :grin:).

**Comment by benfitzpatrick** [comment]
Looks fine to me, tests fine.

**Pull request merged**